In [2]:
import numpy as np
from scipy import stats
from scipy.stats import norm

5.	In January 2005, a company that monitors Internet traffic (WebSideStory) reported that its sampling revealed that the Mozilla Firefox browser launched in 2004 had grabbed a 4.6% share of the market.

I.	If the sample were based on 2,000 users, could Microsoft conclude that Mozilla has a less than 5% share of the market?


In [11]:
# Null Hypothesis
# p >= 5%

# Alternate Hypothesis
# p < 5%

# As alternate hypothesis is having less than symbol (<) so it is a left tail test.

In [13]:
# Confidence Interval
ci = 0.95
alpha = 1 - ci
alpha

0.050000000000000044

In [8]:
z_stat=(0.046-0.05)/(np.sqrt((0.05*(1-0.05))/2000))
z_stat

-0.820782681668124

In [14]:
# For Two Tail Test :
# Z Critical = ppf(1-alpha/2)

# For Left Tail Test :
# Z Critical = ppf(alpha)

# For Right Tail Test :
# Z Critical = ppf(1-alpha)

z_critical = stats.norm.ppf(alpha)
z_critical

-1.6448536269514722

In [10]:
z_critical < z_stat

True

Inference - Here, Null Hypothesis is accepted..!
            Mozilla market share >= 5% Thus, Microsoft can not conclude that Mozilla has a less than 5% share of the market.

## II.	WebSideStory claims that its sample includes all the daily Internet users. If that’s the case, then can Microsoft conclude that Mozilla has a less than 5% share of the market

As WebSideStory claims that its sample includes all daily internet users so their sample is very close to the population. So Microsoft can accept the statistics of WebSideStory that is Mozilla share is 4.6%. Hence Microsoft can say that Mozilla has less than 5% share of the market.

Questions 8 and 9 are based on the following: To prepare a report on the economy, analysts need to estimate the percentage of businesses that plan to hire additional employees in the next 60 days.



8.	How many randomly selected employers (minimum number) must we contact in order to guarantee a margin of error of no more than 4% (at 95% confidence)?



In [6]:
# My P-value would be 0.5
#As margin of error is given not more than 4% i.e. = 0.04
# The Z value for the 95% confidenc interval is (-1.6)
#Probablity of not getting selected Q = 1 - P = 1 - 0.5 = 0.5

#Now, Sample Size Formula is 
# S = Z^2 * P * Q / E^2

# ci is given and we need to calculate z score

import scipy.stats as stats
a = stats.norm.ppf(0.95)

S = ((a)**2 * 0.5 * 0.5)/(0.04)**2
S

422.74116470240824

In [7]:
#If we take the value manually from the Z-Table then 
S = ((-1.6)**2  * 0.5 * 0.5)/(0.04)**2
S

400.00000000000006

So, answer is Option B)

9.	Suppose we want the above margin of error to be based on a 98% confidence level. What sample size (minimum) must we now use?



In [8]:
# My P-value would be 0.5
#As margin of error is given not more than 4% i.e. = 0.04
# The Z value for the 95% confidenc interval is (-1.6)
#Probablity of not getting selected Q = 1 - P = 1 - 0.5 = 0.5

#Now, Sample Size Formula is 
# S = Z^2 * P * Q / E^2

# ci is given and you need to calculate z score


In [9]:
# z score is given and we need to calculate ci
stats.norm.cdf(2.05)

0.9798177845942956

In [10]:
stats.norm.ppf(0.95)

1.6448536269514722

In [11]:
stats.norm.ppf(0.98) * stats.norm.ppf(0.98) * 0.5 * 0.5 / (0.04)**2

659.0444668627183

Answer is Option B), AS 659 is more close to 757.